pkg/csi/service/wcp/controller_helper.go

```diff
@@ -330,34 +335,40 @@ func getDatastoreURLFromStoragePool(ctx context.Context, spName string) (string,
 	return datastoreURL, nil
 }
 
-// getAccessibleNodesFromStoragePool returns the accessibleNodes pertaining to the given StoragePool
-func getAccessibleNodesFromStoragePool(ctx context.Context, spName string) ([]string, error) {
+// getStoragePoolInfo returns the accessibleNodes and the storage-pool-type pertaining to the given StoragePool
+func getStoragePoolInfo(ctx context.Context, spName string) ([]string, string, error) {
 	// Get a config to talk to the apiserver
 	cfg, err := config.GetConfig()
 	if err != nil {
-		return nil, fmt.Errorf("failed to get Kubernetes config. Err: %+v", err)
+		return nil, "", fmt.Errorf("failed to get Kubernetes config. Err: %+v", err)
 	}
 
 	// create a new StoragePool client
 	spClient, err := dynamic.NewForConfig(cfg)
 	if err != nil {
-		return nil, fmt.Errorf("failed to create StoragePool client using config. Err: %+v", err)
+		return nil, "", fmt.Errorf("failed to create StoragePool client using config. Err: %+v", err)
 	}
 	spResource := spv1alpha1.SchemeGroupVersion.WithResource("storagepools")
 
 	// Get StoragePool with spName
 	sp, err := spClient.Resource(spResource).Get(ctx, spName, metav1.GetOptions{})
```

Comment:
Ideally we shouldn't make any Kubernetes calls from controller.go and controller_helper.go. This is ensure that the controller level is C0 agnostic.

You should expose and interface here - https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/csi/service/common/commonco/coagnostic.go and let the https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go implement it.

Can you make this change.

Reply:
this is pre-existing code not need code. Also as far as I understand this is actually a grpc call out of the csi driver to keep the driver CO agnostic. @shreesha is the original author of this code and he can comment more.

Reply:
I think @subramanian-neelakantan is the original author for the [dynamic client](https://github.com/kubernetes/client-go/tree/master/examples/dynamic-create-update-delete-deployment). He did it while implementing storage pools and I think he would be the right person to comment on this.

Reply:
Since this change does not add or alter this functionality we can take this discussion offline to decide if this code needs to change. But resolving the comment since it is not about my code changes.